.github/workflows/docs.yaml

```diff
@@ -0,0 +1,51 @@
+name: Build Docs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: docs-deployment-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  build-deploy:
+    name: Build & Deploy Docs
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Python Dependencies
+        uses: HassanAbouelela/actions/setup-python@setup-python_v1.0.0
```

Comment:
Isn't there an official setup-python we can use here?

Reply:
The setup-python action just sets up python, while the one from my actions repo does a bit more under the hood to set up poetry and caching.

There is a bit more context for why this is on my private account in dev-core, but this is essentially something we're planning on moving to everywhere eventually, pending a few things.